src/test/codegen/match-optimizes-away.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+//
+// no-system-llvm
+// compile-flags: -O
+#![crate_type="lib"]
+
+pub enum Three { First, Second, Third }
+use Three::*;
+
+pub enum Four { First, Second, Third, Fourth }
+use Four::*;
```

Comment:
It'd be nice if we made a lint against this, set to error at least for the Rust codebase...

Reply:
wow thanks for catching this, I had them in 2 separate files before & just assumed that any warning would trap here...

Reply:
There are warnings from the `match`es themselves but most tests are written without warning avoidance in mind so we'd have to do *a lot* of work to ban warnings from tests.

Reply:
Should probably as a matter of course put `#![deny(warnings)]` on these tests where it is appropriate (at least I will do that in future)